src/Simplex/Chat.hs

```diff
@@ -3225,9 +3227,8 @@ receiveViaCompleteFD user fileId RcvFileDescr {fileDescrText, fileDescrComplete}
     getUnknownSrvs :: [XFTPServer] -> CM [XFTPServer]
     getUnknownSrvs srvs = do
       ChatConfig {defaultServers = DefaultAgentServers {xftp = defXftp}} <- asks config
-      storedSrvs <- map (\ServerCfg {server} -> protoServer server) <$> withStore' (`getProtocolServers` user)
-      let defXftp' = L.map protoServer defXftp
-          knownSrvs = fromMaybe defXftp' $ nonEmpty storedSrvs
+      storedSrvs <- withStore' (`getProtocolServers` user)
+      let knownSrvs = L.map (\ServerCfg {server} -> protoServer server) $ fromMaybe defXftp $ nonEmpty storedSrvs
```

Comment:
I think we should refactor shared logic of retrieving user "used" servers that reads server list from db, takes default from config, and returns default if list from db is empty. there are several places (`getUnknownSrvs`, `GetAgentServersSummary`, `APIGetUserProtoServers`) that all do basically this same thing but in different ways and it's hard to find common pattern in what they're doing if you don't know what you're looking for